one-on-one-agent.py

```diff
@@ -125,56 +122,56 @@ def generate_summary(memory: ConversationBufferMemory) -> str:
     
     Summary:
     """
-    
+
     summary = chat_llm.invoke(prompt)
     return summary.content
 
+
 def process_input(state: AgentState) -> AgentState:
     messages = state["messages"]
     memory = state["memory"]
     update_state = state["update_state"]
-    
+
     human_input = messages[-1].content if isinstance(messages[-1], HumanMessage) else ""
     state["last_human_message"] = human_input
     memory.chat_memory.add_user_message(human_input)
-    
+
     # We don't generate an AI response here anymore
     # The AI response will be handled in the check_update function
-    
+
     return {
         **state,
         "memory": memory,
         "update_state": update_state,
     }
 
+
 def check_update(state: AgentState) -> AgentState:
     update_state = state["update_state"]
     last_human_message = state["last_human_message"]
     memory = state["memory"]
-    
+
     # Use LLM to categorize the input
     category, detected_value = categorize_input(last_human_message)
-    
+
     # Update the update_state based on the LLM categorization
     if category in update_state:
         update_state[category] = True
-    
+
     # Handle email detection
-    if category == "email" and detected_value:
+    if category == "email" and detected_value and state["user_email"] == None:
         update_state["email"] = True
         state["user_email"] = detected_value
-    elif not update_state.get("email") and detected_value:
```

Comment:
removing this `elif` because I don't think it was helping with getting/setting the email and I think we only needed the if statement above